### PR TITLE
accordion

### LINKED ISSUE #14: Accordion block
https://www.bevespi.com/how-to-use-bevespi-inhaler.html

Screenshot 2023-10-05 at 13 58 37

### PR DESCRIPTION
**Pull request by lansami**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #14 

Test URLs:
- Before: https://main--bevespi--hlxsites.hlx.page/how-to-use-bevespi-inhaler
- After: https://issue-14-accordion--bevespi--hlxsites.hlx.page/how-to-use-bevespi-inhaler

### CODE REVIEW

File: blocks/accordion/accordion.css [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+.accordion .accordion-row.expanded .accordion-row-content p.item-with-picture-before picture img {
+    width: 35px;
+    padding-right: 12px;
+    max-width: unset;
+}
+
+.accordion .accordion-row .accordion-row-content .expandable-button {
+    position: absolute;
```
Comment (hannessolo) [inline blocks/accordion/accordion.css:83]: can you add `cursor: pointer` to this button?

**Review by hannessolo — COMMENTED** [review]
Screenshot 2023-10-18 at 16 38 19

I would probably recommend fixing this overlap, even though it is also like this on the real site.

File: blocks/accordion/accordion.css (hunk 1 above)
Comment (lansami) [reply blocks/accordion/accordion.css:83]: fixed the issue, had to add `pointer-events: auto` because it was not showing the pointer

**Review by paolomoz — COMMENTED** [review]
@lansami it looks quite good! see below some adjustments:
- "More" button is too heavy, make it 500
- reduce padding bottom (within the box) by 4px
- when click on "More +" , content shifts some pixels to the top
- margin between boxes is 10px not 13px

CLS is a concern here. Regardless the ISI sticky, the accordion generates 0.074 of CLS
https://pagespeed.web.dev/analysis/https-issue-14-accordion--bevespi--hlxsites-hlx-page-how-to-use-bevespi-inhaler/g72l1zwygp?form_factor=mobile

More general comment: this is a lot of code. Why do we need 300+ LOC for this block?

**Comment by lansami** [comment]
> @lansami it looks quite good! see below some adjustments:
> 
> * "More" button is too heavy, make it 500
> * reduce padding bottom (within the box) by 4px
> * when click on "More +" , content shifts some pixels to the top
> * margin between boxes is 10px not 13px
> 
> CLS is a concern here. Regardless the ISI sticky, the accordion generates 0.074 of CLS https://pagespeed.web.dev/analysis/https-issue-14-accordion--bevespi--hlxsites-hlx-page-how-to-use-bevespi-inhaler/g72l1zwygp?form_factor=mobile
> 
> More general comment: this is a lot of code. Why do we need 300+ LOC for this block?

The huge number of LOC are because there are some specific cases that required dom changing and the css file has a lot of LOC because of the display cases for this type of accordion
The display of pictures in line with some text required a wrapper because the text may contain the `strong` tag.
The header contains the first paragraph for the cases when there is a preview of the content.
The click event has two actions to be done, one to toggle the display of the accordion item content and another to close all other accordion items
I've set some fixed heights or min-heights or min-width for pictures and some containers, in order to render them directly with a specified dimension, to not generate CLS
The use of pseudoelements increased the number of lines also

**Review by paolomoz — APPROVED** [review]

**Pull request merged**